### PR TITLE
[benchmark] Add query_kernel_profiler_() and benchmark/misc for performance monitoring

### PR DESCRIPTION
**Pull request by yolo2themoon**
Related issue = #948

### CODE REVIEW

File: taichi/program/program.h [hunk 1]
```diff
@@ -134,6 +134,23 @@ class Program {
     profiler->print();
   }
 
+  struct KernelProfilerQueryResult {
+    int counter;
```
Comment (k-ye) [inline taichi/program/program.h:138]:
```suggestion
    int counter{0};
```

File: taichi/program/program.h [hunk 2]
```diff
@@ -134,6 +134,23 @@ class Program {
...
+    double min;
+    double max;
+    double avg;
+
+    KernelProfilerQueryResult() : counter(0), min(0), max(0), avg(0) {
+    }
+  };
+
+  KernelProfilerQueryResult kernel_profiler_query(const std::string &name) {
```
Comment (k-ye) [inline taichi/program/program.h:147]:
```suggestion
  KernelProfilerQueryResult query_kernel_profiler(const std::string &name) {
```

**Review by k-ye — APPROVED** [review]
LGTM!

File: taichi/program/program.h (hunk 1 above)
Comment (yolo2themoon) [reply taichi/program/program.h]: 5dbfd48e6bd64642d7c5205320e5fddaae197dcb

File: taichi/program/program.h (hunk 2 above)
Comment (yolo2themoon) [reply taichi/program/program.h]: 5dbfd48e6bd64642d7c5205320e5fddaae197dcb

**Pull request merged**